framework.py

```diff
@@ -9,16 +9,150 @@
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
 import sgtk
+from sgtk.util import filesystem
+
+import threading
+import datetime
+import os
+import time
 
 class ShotgunUtilsFramework(sgtk.platform.Framework):
     
+    # A list of file names which should never be deleted when cleaning up old
+    # cached data.
+    _ALWAYS_KEEP_CACHED_FILES = [
+        "sg_schema.pickle",
+        "sg_status.pickle",
+    ]
     ##########################################################################################
     # init and destroy
             
     def init_framework(self):
+        """
+        Init this framework.
+
+        Post an old cached data cleanup in the background
+        """
         self.log_debug("%s: Initializing..." % self)
+        self._stop_cleanup = False
+        self._bg_cleanup_thread = None
+        self._post_old_data_cleanup()
     
     def destroy_framework(self):
+        """
+        Destroy this framework.
+
+        If an old cached data cleanup was posted in the background, stop it
+        immediately.
+        """
         self.log_debug("%s: Destroying..." % self)
+        # Please note that we are modifying a member which is read in another
+        # thread which should be fine in Python with the GIL protecting its access.
+        self._stop_cleanup = True
+        if self._bg_cleanup_thread:
+            if self._bg_cleanup_thread.isAlive():
+                # If the clean up is not completed yet, log why we are waiting.
+                self.log_info("Waiting for old data clean up to complete...")
+            self._bg_cleanup_thread.join()
```

Comment:
nitpick: do we need to wait for the thread if it is not alive? should this line be inside the if clause?

Reply:
I think it is good practice to always join threads (or processes). The `if` is here only to log something if we think `join` will not return immediately. With the _stop cleanup_ hand break I added following your comments, I never saw the log message during my tests.